src/scheduler/agent/database.c

```diff
@@ -1132,7 +1169,38 @@ char* get_email_command(scheduler_t* scheduler, char* user_email)
         g_string_append(client_cmd, "warn");
       }
     }
+    g_string_append_printf(client_cmd, " -S v15-compat");
+    temp_smtpvariable = (char *)g_hash_table_lookup(smtpvariables, "SMTPPort");
+    temp_smtpvariable2 = (char *)g_hash_table_lookup(smtpvariables, "SMTPStartTls");
+    /* use smtps only if port is not 25 or SMTPStartTls is provided */
+    if((g_strcmp0(temp_smtpvariable, "25") !=  0) || g_strcmp0(temp_smtpvariable2, "1") == 0)
+    {
+      g_string_append_printf(client_cmd, " -S mta=\"smtps://");
+    }
+    else
+    {
+      g_string_append_printf(client_cmd, " -S mta=\"smtp://");
+    }
+    if(g_hash_table_contains(smtpvariables, "SMTPAuthUser"))
+    {
+      temp_smtpvariable = (char *)g_hash_table_lookup(smtpvariables, "SMTPAuthUser");
+      temp_smtpvariable2 = url_encode(temp_smtpvariable);
```

Comment:
I was just looking at glib and found this function [g_string_append_uri_escaped()](https://developer.gnome.org/glib/stable/glib-Strings.html#g-string-append-uri-escaped)

It seems to work just fine with `reserved_chars_allowed` marked as `NULL` and `allow_utf8` set as `TRUE`. This is eliminate requirement of additional `url_encode()` function and can save one temporary variable too.